Marlin/SanityCheck.h

```diff
@@ -1019,3 +1019,53 @@
 #if COUNT_LCD_24 > 1
   #error "Please select no more than one LCD controller option."
 #endif
+
+#if ENABLED( SPINDLE_ENABLE)
+  #if !PIN_EXISTS(SPINDLE_ENABLE)
+    #error "SPINDLE_ENABLE requires SPINDLE_ENABLE_PIN."
+  #elif !defined(SPINDLE_ENABLE_INVERT) || !(SPINDLE_ENABLE_INVERT == true || SPINDLE_ENABLE_INVERT == false)
+    #error "SPINDLE_ENABLE_INVERT missing or incorrectly defined"
+  #elif ENABLED (SPINDLE_SPEED) && PIN_EXISTS(SPINDLE_SPEED)
+    #if !((SPINDLE_SPEED_PIN >=2 && SPINDLE_SPEED_PIN <= 13) || (SPINDLE_SPEED_PIN >=44 && SPINDLE_SPEED_PIN <= 46) )
+      #error "SPINDLE_SPEED_PIN not assigned to a PWM pin"
+    #elif !defined(SPINDLE_SPEED_INVERT) || !(SPINDLE_SPEED_INVERT == true || SPINDLE_SPEED_INVERT == false)
+      #error "SPINDLE_SPEED_INVERT missing or incorrectly defined"
+    #elif !defined(RPM_SLOPE) || !defined(RPM_INTERCEPT) || !defined(RPM_MIN) || !defined(RPM_MAX)
+      #error "spindle speed equation constant(s) missing"
+    #elif SPINDLE_SPEED_PIN == 4 || SPINDLE_SPEED_PIN == 11 || SPINDLE_SPEED_PIN == 12 || SPINDLE_SPEED_PIN == 13
+      #error "Counter/Timer for SPINDLE_SPEED PWM pin is used by a system interrupt "
+    #elif PIN_EXISTS(X_MAX) &&  X_MAX_PIN == SPINDLE_SPEED_PIN && defined(USE_XMAX_PLUG)
```

Comment:
We've already established that `SPINDLE_SPEED_PIN` is something other than -1. We also know that `X_MAX_PIN` exists. We also know that if `USE_XMAX_PLUG` is set that `X_MAX_PIN` cannot be -1. So you only need `X_MAX_PIN == SPINDLE_SPEED_PIN` here.

Only one space after `&&`. I will not merge a PR that has even a single space out of place, tabs instead of spaces, or trailing whitespace.